Trustroots/trustroots
# Migrate Search map to React

Migrate the main map search to React.

#### Proposed Changes

* Uses `react-map-gl` for Map components React library to render (already used elsewhere for other maps)
* Instead of reusing base `Map` component, extract its components and re-use to build more custom map component for search
* For now, pass data across Angular/React via props (observe changes in prop values and use callback functions). I didn't want to start adding shared data layer just yet because it could end up complicated across Angular and React. Right next up we should start migrating rest of the search UI to React and it's also a good chance to introduce data layer and dispact/listen actions via it, instead of using props and callbacks.
* Has 3 different Mapbox layers and one OSM layer for development purposes only.
* Convert API response to a valid GeoJson response thus making it more straightforward to use "natively" with mapping libraries 

<img width="1259" alt="Screenshot 2020-12-21 at 16 28 25" src="https://user-images.githubusercontent.com/87168/102891369-bc36e800-4466-11eb-9a92-30be64b73672.png">

<img width="1258" alt="Screenshot 2020-12-21 at 16 27 14" src="https://user-images.githubusercontent.com/87168/102891378-c0fb9c00-4466-11eb-9627-56d1e0916941.png">

<img width="196" alt="Screenshot 2020-12-21 at 16 27 56" src="https://user-images.githubusercontent.com/87168/102891372-be00ab80-4466-11eb-8637-5f1171c1275a.png">

<img width="1080" alt="Screenshot 2021-01-09 at 18 23 40" src="https://user-images.githubusercontent.com/87168/104102971-4923d300-52a8-11eb-996e-22ed34eeddc9.png">


#### Testing

Test `/search` thoroughly:

* Configure Mapbox token. Config & token can be copied from `config/env/local.sample.js` and place in `config/env/local.js`

* Search for locations
* Click "see in big map" from profiles to open `/search` at correct location
* Click on location names from profiles to open search at right location
* Change map layers
* Try Zooming and browsing the map with buttons, mouse scroll/drag, touch device if possible
* Refresh the page and observe map remember location and map layer
* Open offers, note how already opened dots stay faded
* Click clusters to zoom into them
* Ensure "meet" offers work
* Ensure different filters work
* Ensure mobile experience works
* Try search map with user who didn't confirm their email yet (freshly created user). No API queries should fire.
